invokeai/app/services/shared/sqlite/migrations/migration_2.py

```diff
@@ -0,0 +1,97 @@
+import sqlite3
+
+from invokeai.app.services.shared.sqlite.sqlite_migrator import Migration
+
+
+def _migrate(cursor: sqlite3.Cursor) -> None:
+    """Migration callback for database version 2."""
+
+    _add_images_has_workflow(cursor)
+    _add_session_queue_workflow(cursor)
+    _drop_old_workflow_tables(cursor)
+    _add_workflow_library(cursor)
+    _drop_model_manager_metadata(cursor)
+
+
+def _add_images_has_workflow(cursor: sqlite3.Cursor) -> None:
+    """Add the `has_workflow` column to `images` table."""
+    cursor.execute("ALTER TABLE images ADD COLUMN has_workflow BOOLEAN DEFAULT FALSE;")
+
+
+def _add_session_queue_workflow(cursor: sqlite3.Cursor) -> None:
+    """Add the `workflow` column to `session_queue` table."""
+    cursor.execute("ALTER TABLE session_queue ADD COLUMN workflow TEXT;")
+
+
+def _drop_old_workflow_tables(cursor: sqlite3.Cursor) -> None:
+    """Drops the `workflows` and `workflow_images` tables."""
+    cursor.execute("DROP TABLE workflow_images;")
+    cursor.execute("DROP TABLE workflows;")
+
+
+def _add_workflow_library(cursor: sqlite3.Cursor) -> None:
+    """Adds the `workflow_library` table and drops the `workflows` and `workflow_images` tables."""
+    tables = [
+        """--sql
+        CREATE TABLE workflow_library (
```

Comment:
I wonder if this and similar methods should use `CREATE TABLE IF NOT EXISTS` instead of the unconditional `CREATE TABLE`. This would future proof against a crash at a critical moment when some of the tables have been added and others aren't.

Reply:
I've gone ahead and done this for this particular migration, but I've improved the resiliency of the migrator and don't think we need this kind of idempotency for migrations - we rely on not running a migration multiple times for that.